operator/pkg/values/translation/translation.go

```diff
@@ -138,40 +138,88 @@ func GetGlobalMonitoring(m *platform.GlobalMonitoring) *ValuesBuilder {
-			}
+// SetScannerComponentDisabled sets the disabled values for scanner configurations
+func SetScannerComponentDisabledValue(sv *ValuesBuilder, scannerComponent *platform.ScannerComponentPolicy) {
+	if scannerComponent != nil {
+		switch *scannerComponent {
+		case platform.ScannerComponentDisabled:
+			sv.SetBoolValue("disable", true)
+		case platform.ScannerComponentEnabled:
+			sv.SetBoolValue("disable", false)
+		default:
+			sv.SetError(fmt.Errorf("invalid spec.scanner.scannerComponent %q", *scannerComponent))
 		}
-		autoscaling.SetInt32("minReplicas", scaling.MinReplicas)
-		autoscaling.SetInt32("maxReplicas", scaling.MaxReplicas)
-		sv.AddChild("autoscaling", &autoscaling)
 	}
+}
 
-	if analyzer != nil {
-		sv.SetStringMap("nodeSelector", analyzer.NodeSelector)
-		sv.AddChild(ResourcesKey, GetResources(analyzer.Resources))
-		sv.AddAllFrom(GetTolerations(TolerationsKey, analyzer.DeploymentSpec.Tolerations))
+// SetScannerAnalyzerValues sets values in "sv" based on "analyzer".
+func SetScannerAnalyzerValues(sv *ValuesBuilder, analyzer *platform.ScannerAnalyzerComponent) {
+	if analyzer == nil {
+		return
 	}
+	setScannerComponentScaling(sv, analyzer.GetScaling())
+	sv.SetStringMap("nodeSelector", analyzer.NodeSelector)
+	sv.AddChild(ResourcesKey, GetResources(analyzer.Resources))
+	sv.AddAllFrom(GetTolerations(TolerationsKey, analyzer.DeploymentSpec.Tolerations))
 }
 
-// SetScannerDBValues sets values in "sb" based on "db".
+// SetScannerDBValues sets values in "sv" based on "db".
 func SetScannerDBValues(sv *ValuesBuilder, db *platform.DeploymentSpec) {
 	if db != nil {
 		sv.SetStringMap("dbNodeSelector", db.NodeSelector)
 		sv.AddChild("dbResources", GetResources(db.Resources))
 		sv.AddAllFrom(GetTolerations("dbTolerations", db.Tolerations))
 	}
 }
+
+// SetScannerV4DBValues sets values in "sv" based on "db"
+func SetScannerV4DBValues(sv *ValuesBuilder, db *platform.ScannerV4DB) {
+	if db == nil {
+		return
+	}
+
+	dbVB := NewValuesBuilder()
+	dbVB.SetStringMap("nodeSelector", db.NodeSelector)
+	dbVB.AddChild(ResourcesKey, GetResources(db.Resources))
+	dbVB.AddAllFrom(GetTolerations(TolerationsKey, db.Tolerations))
+	// TODO(ROX-19051): translate persistence values
+	sv.AddChild("db", &dbVB)
+	return
+}
+
+// SetScannerV4ComponentValues sets values in "sv" based on "component"
+func SetScannerV4ComponentValues(sv *ValuesBuilder, componentKey string, component *platform.ScannerV4Component) {
+	if component == nil {
+		return
+	}
+
+	componentVB := NewValuesBuilder()
+	setScannerComponentScaling(sv, component.Scaling)
+	sv.SetStringMap("nodeSelector", component.NodeSelector)
+	sv.AddChild(ResourcesKey, GetResources(component.Resources))
+	sv.AddAllFrom(GetTolerations(TolerationsKey, component.Tolerations))
+	sv.AddChild(componentKey, &componentVB)
+}
+
+func setScannerComponentScaling(sv *ValuesBuilder, scaling *platform.ScannerComponentScaling) {
+	if scaling == nil {
+		return
+	}
+
+	sv.SetInt32("replicas", scaling.Replicas)
```

Comment:
Do we have any normalization/sanitization going on somewhere? I'm wondering about invalid values for the CR fields, e.g. a negative number of replicas or something.

Reply:
We have validation at the CR level e.g https://github.com/stackrox/stackrox/blob/aaf511abb67f4e024f3d402e52cc38e23edefc23/operator/apis/platform/v1alpha1/common_types.go#L160-L164